src/lab1/structures/vector/Vector3D.ts

```diff
@@ -11,6 +15,16 @@ export default class Vector3D {
     this.z = z;
   }
 
+  rotate(angleX: number, angleY: number, angleZ: number): Vector3D {
+    return transformVector(
+      this,
+      transformations.rotate3d(angleX, angleY, angleZ)
+    );
+  }
+  scale(x: number, y: number, z: number): Vector3D {
+    return transformVector(this, transformations.scale3d(x, y, z));
+  }
+
```

Comment:
Access modifiers?
Why can't we translate Vector? Also it should implement Transformable interface

Reply:
Added translate, but Vector (and Vertex) don't implement Transformable because Transformable only returns void. I.e. they modify their internal structure in place. Whereas Vector and Vertex shouldn't do this, they should be immutable, because for example many triangles might depend on one Vertex, and otherwise when you transform one triangle by transforming Vertex you'd accidentally also transform adjacent triangles. 

I tried changing Transformable interface so that it has to return a reference to a new object, then I could cover both immutable and mutable patterns (by simply returning reference to `this`), but abandoned this idea as code gets too complicated and we don't really benefit from this abstraction, i.e. we don't need to hide Vectors and Vertices behind this interface currently.

Reply:
On the second thought, I removed translate from Vector. This doesn't make sense with it, I think if you have to translate vectors (not vertices) you're likely doing something wrong